srcCpp/RTIDDSImpl.cxx

```diff
@@ -917,6 +908,17 @@ class RTIPublisher : public IMessagingWriter
     RTIPublisher(DDSDataWriter *writer, unsigned long num_instances, RTIOsapiSemaphore * pongSemaphore, int instancesToBeWritten)
     {
         _writer = T::DataWriter::narrow(writer);
+        if (_writer == NULL) {
+            fprintf(stderr, "DataWriter::narrow(writer) error.\n");
+            /*
+             * Nothing was created at this point, shutdown() is called for
+             * maintain consistency.
+             */
+            Shutdown();
```

Comment:
_instance_handles hasn't been initialized when Shutdown calls free on it. If _instance_handles is garbage, free() can crash. 

Initialize _instance_handles to NULL or don't call Shutdown().